RevenueCatUI/Templates/V2/Components/Image/ImageComponentView.swift

```diff
@@ -52,20 +54,31 @@ struct ImageComponentView: View {
                     darkUrl: style.darkUrl,
                     darkLowResUrl: style.darkLowResUrl
                 ) { (image, size) in
-                    self.renderImage(image, size, with: style)
+                    self.renderImage(image, size, maxWidth: maxWidth, with: style)
                 }
-                .size(style.size)
+                .applyImageWidth(size: style.size)
+                .applyImageHeight(size: style.size, aspectRatio: self.aspectRatio(style: style))
                 .clipped()
                 .padding(style.padding.extend(by: style.border?.width ?? 0))
                 .shape(border: style.border,
                        shape: style.shape)
                 .shadow(shadow: style.shadow,
                         shape: style.shape?.toInsettableShape())
                 .padding(style.margin)
+                .onWidthChange { width in
+                    self.maxWidth = self.calculateMaxWidth(parentWidth: width, style: style)
+                }
             }
         }
     }
 
+    private func calculateMaxWidth(parentWidth: CGFloat, style: ImageComponentStyle) -> CGFloat {
+        let totalBorderWidth = (style.border?.width ?? 0) * 2
+        return parentWidth - totalBorderWidth
+            - style.margin.leading - style.margin.trailing
+            - style.padding.leading - style.padding.trailing
```

Comment:
Added padding here as it also affects the width of the image

Reply:
Is this being called anywhere? Should it maybe be applied to the GeometryReader on line 70 to prevent it from taking up all available space?